src/providers/ses.rs

```diff
@@ -44,44 +46,70 @@ impl Provider for SesProvider {
         subject: &str,
         body_text: &str,
         body_html: Option<&str>,
     ) -> Result<String, ProviderError> {
-        let mut destination = Destination::default();
-        destination.to_addresses = Some(vec![to.to_string()]);
+        let mut all_headers = header::Headers::new();
+        all_headers.set(header::From(vec![self.sender.parse().unwrap()]));
+        all_headers.set(header::To(vec![to.parse().unwrap()]));
+        all_headers.set(header::Subject(subject.into()));
+        all_headers.set(header::MIME_VERSION_1_0);
+
         if cc.len() > 0 {
-            destination.cc_addresses = Some(cc.iter().map(|s| s.to_string()).collect());
+            let mut parsed_cc: Vec<Mailbox> = Vec::new();
+            for address in cc.iter() {
+                parsed_cc.push(address.parse().unwrap())
+            }
+            all_headers.set(header::Cc(parsed_cc))
+        }
+
+        if let Some(headers) = headers {
+            for (name, value) in headers {
+                all_headers.append_raw(name.to_owned(), value.to_owned())
+            }
         }
 
-        let mut message = Message::default();
-        message.subject.data = subject.to_string();
-        message.body.text = Some(Content {
-            charset: None,
-            data: body_text.to_string(),
-        });
-        if let Some(html) = body_html {
-            message.body.html = Some(Content {
-                charset: None,
-                data: html.to_string(),
-            });
+        let mut message: Message<MultiPart<String>> = Message::new().with_headers(all_headers);
+
+        let mut body: MultiPart<String> = MultiPart::alternative().with_singlepart(
+            SinglePart::quoted_printable()
+                .with_header(header::ContentType(
+                    "text/plain; charset=utf8".parse().unwrap(),
+                ))
+                .with_body(body_text.to_owned()),
+        );
+        if let Some(body_html) = body_html {
+            body = body.with_multipart(
+                MultiPart::related().with_singlepart(
+                    SinglePart::eight_bit()
+                        .with_header(header::ContentType(
+                            "text/html; charset=utf8".parse().unwrap(),
+                        ))
+                        .with_body(body_html.to_owned()),
+                ),
+            )
         }
 
-        let mut request = SendEmailRequest::default();
-        request.destination = destination;
-        request.message = message;
-        request.source = self.sender.to_string();
+        message = message.with_body(MultiPart::mixed().with_multipart(body));
+
+        let encoded_message = encode(&format!("{}", message));
+        let mut request = SendRawEmailRequest::default();
+        request.raw_message = RawMessage {
+            data: format!("{}", encoded_message).as_bytes().to_vec(),
+        };
 
         self.client
-            .send_email(&request)
+            .send_raw_email(&request)
```

Comment:
with the amount of changes to switch to RawMessage to support headers I think we should land this PR and then spin up a different issue / PR to finalize the ALLOW_LIST passthrough for the known headers (from https://github.com/mozilla/fxa-email-service/issues/75#issuecomment-398420128 )

Reply:
yes that's what i was thinking.